tests/spec/w3c/headers-spec.js

```diff
@@ -557,6 +557,105 @@ describe("W3C — Headers", () => {
+
+      const doc = await makeRSDoc(ops);
+
+      const h1Elem = doc.querySelector("h1#title");
+      // html is not escaped
+      expect(h1Elem.childNodes[1].localName).toBe("code");
+      expect(h1Elem.textContent).toBe("Spec Marked Up Level 0");
+      expect(doc.title).toBe("Spec Marked Up Level 0");
+
+      const terms = doc.querySelectorAll("dt");
+      expect(terms[0].textContent).toBe("This version:");
+      expect(terms[0].nextElementSibling.localName).toBe("dd");
+      expect(terms[0].nextElementSibling.textContent).toContain("REC-abc-0");
+      expect(terms[1].textContent).toBe("Latest published version:");
+      expect(terms[1].nextElementSibling.localName).toBe("dd");
+      expect(terms[1].nextElementSibling.textContent).toContain("TR/abc-0/");
+    });
+
+    it("updates relevant locations with the level when level is an integer greater than 0", async () => {
+      const body = `
+      <h1 id="title">Spec <code>Marked</code> Up</h1>${makeDefaultBody()}`;
+      const ops = makeStandardOps({}, body);
+      const newProps = { level: 9870, specStatus: "REC", shortName: "xyz" };
+      Object.assign(ops.config, newProps);
+
+      const doc = await makeRSDoc(ops);
+
+      const h1Elem = doc.querySelector("h1#title");
+      // html is not escaped
+      expect(h1Elem.childNodes[1].localName).toBe("code");
+      expect(h1Elem.textContent).toBe("Spec Marked Up Level 9870");
+      expect(doc.title).toBe("Spec Marked Up Level 9870");
+
+      const terms = doc.querySelectorAll("dt");
+      expect(terms[0].textContent).toBe("This version:");
+      expect(terms[0].nextElementSibling.localName).toBe("dd");
+      expect(terms[0].nextElementSibling.textContent).toContain("REC-xyz-9870");
+      expect(terms[1].textContent).toBe("Latest published version:");
+      expect(terms[1].nextElementSibling.localName).toBe("dd");
+      expect(terms[1].nextElementSibling.textContent).toContain("TR/xyz-9870/");
+    });
+  });
+
+  describe("invalid level configs", () => {
+    it("warns the user and does not add the level to the relevant places when level is a string that doesn't convert to an integer", async () => {
+      const body = `
+        <h1 id="title">Spec <code>Marked</code> Up</h1>${makeDefaultBody()}`;
+      const ops = makeStandardOps({}, body);
+      const newProps = { level: "a1", specStatus: "REC", shortName: "xxx" };
+      Object.assign(ops.config, newProps);
+      const doc = await makeRSDoc(ops);
+
+      const h1Elem = doc.querySelector("h1#title");
+      expect(h1Elem.classList).toContain("respec-offending-element");
+      expect(h1Elem.textContent).toBe("Spec Marked Up");
+
+      const date = new Date();
+
+      const terms = doc.querySelectorAll("dt");
+      expect(terms[0].textContent).toBe("This version:");
+      expect(terms[0].nextElementSibling.localName).toBe("dd");
+      expect(terms[0].nextElementSibling.textContent).toContain(
+        `REC-xxx-${date.getFullYear()}`
```

Comment:
We probably don't need a date here. Can use a regex instead if we want the date part

Reply:
I need the date part because if level is added, it becomes `REC-shortname-level-YYYYMMDD`. I'll look into a regex. ^^